src/compiler/utilities.rs

```diff
@@ -1,19 +1,71 @@
 #![allow(non_upper_case_globals)]
 
+use std::cmp::Ordering;
+
 use crate::{BaseNode, DiagnosticMessage, DiagnosticWithDetachedLocation, SyntaxKind};
 
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum OperatorPrecedence {
+    Comma,
+    Multiplicative,
+    Primary,
+    Invalid = -1,
+}
+
+impl OperatorPrecedence {
+    pub const Lowest: OperatorPrecedence = OperatorPrecedence::Comma;
```

Comment:
This seems like a nice way of translating the pattern in the Typescript code where they have "marker" enum values that refer to one of the other enum values (so here I can say `OperatorPrecedence::Lowest` which is of type `OperatorPrecedence` even though it's not actually one of the `OperatorPrecedence` enum variants)